Source/Shaders/OctahedralProjectionAtlasFS.glsl

```diff
@@ -0,0 +1,68 @@
+varying vec2 v_textureCoordinates;
+
+uniform float originalSize;
+uniform sampler2D texture0;
+uniform sampler2D texture1;
+uniform sampler2D texture2;
+uniform sampler2D texture3;
+uniform sampler2D texture4;
+uniform sampler2D texture5;
+
+const float yMipLevel1 = 1.0 - (1.0 / pow(2.0, 1.0));
+const float yMipLevel2 = 1.0 - (1.0 / pow(2.0, 2.0));
+const float yMipLevel3 = 1.0 - (1.0 / pow(2.0, 3.0));
+const float yMipLevel4 = 1.0 - (1.0 / pow(2.0, 4.0));
+
+void main()
+{
+    vec2 uv = v_textureCoordinates;
+    vec2 textureSize = vec2(originalSize * 1.5 + 2.0, originalSize);
+    vec2 pixel = 1.0 / textureSize;
+
+    float mipLevel = 0.0;
+
+    if (uv.x - pixel.x > (textureSize.y / textureSize.x)) {
+        mipLevel = 1.0;
```

Comment:
This if/else chain _feels_ like it can much shorter and use less branching. Since atlas generation is a one time cost it doesn't matter too much, but worth trying something.